docs/design-documents/hal/0004-pin-names-general-guidelines.md

```diff
@@ -0,0 +1,156 @@
+
+    // Px_xx relates to the processor pin connected to the LED
+    #define LED1 = Px_xx  // LED1
+    #define LED2 = Px_xx  // LED2  
+    #define LED3 = Px_xx  // LED3  
+    #define LED4 = Px_xx  // LED4  
+    .  
+    .  
+    #define LEDN = Px_xx   // LEDN
+
+**Using LEDs at application**
+
+The detection of available LEDs at application level can be done as follow:
+
+    #ifdef(LED1)
+        DigitalOut myLED(LED1);
+        myLED = 1;
+    #endif 
+
+Alternatively, if the usage of an LED is required, then the application can detect its availability and generate an error accordingly:
+
+    #ifndef(LED1)
+        #error This application requires the availability of an LED
+    #endif 
+
+It's possible to define new names of LEDs related to its properties, like color or functionality. They can be defined as aliases in the application as shown below:
+
+    #define RED_LED    LED1
+    #define STATUS_LED LED2 
+
+However, these names do not apply to all boards and hence should not be used in official example applications developed by Arm nor Partners, as these are considered platform agnostic.
+
+### Buttons
+
+**Definition of buttons**
+
+Only add buttons that are available on the board. This is an example on how to define buttons in PinNames.h:
+
+    // Px_xx relates to the processor pin connected to the button  
+    #define BUTTON1 = Px_xx  // BUTTON1  
+    #define BUTTON2 = Px_xx  // BUTTON2  
+    #define BUTTON3 = Px_xx  // BUTTON3  
+    #define BUTTON4 = Px_xx  // BUTTON4   
+    .  
+    .  
+    #define BUTTONN = Px_xx   // BUTTONN  
+
+**Using Buttons at application**
+
+The detection of available buttons at application level can be done as follow:
+
+    #ifdef(BUTTON1)
+        DigitalIn myBUTTON(BUTTON1);
+        int input = myBUTTON.read();
+    #endif 
+
+Alternatively, if the usage of a button is required, then the application can detect its availability and generate an error accordingly:
+
+    #ifndef(BUTTON1)
+        #error This application requires the availability of a button
+    #endif 
+
+It's possible to define new names of buttons related to its properties. It's recommended to add a comment with a description. They can be defined as aliases at application level as shown below:
+
+    #define PUSH_BUTTON BUTTON1 // Momentary push Button
+
+However, these names do not apply to all boards and hence should not be used in official example applications developed by Arm nor Partners, as these are considered platform agnostic.
+
+### UART
+
+Every Mbed board includes a serial interface to the host PC, which allows the console to print useful information about the application.
+
+The UART pins must be defined to be able to run automated tests using Greentea, as the communication between the host PC and the MCU is done over serial.
+
+This is an example on how to define UART names in PinNames.h:
+
+    // Px_xx relates to the processor pin connected to the UART
+    USBTX       = Px_xx,
```

Comment:
As my above comment, is this a macro or enum value?

Reply:
I've updated the doc to clarify that USBx continue to use enums